src/table/boot.rs

```diff
@@ -128,17 +128,32 @@ pub struct BootServices {
     check_event: unsafe extern "efiapi" fn(event: Event) -> Status,
 
     // Protocol handlers
-    install_protocol_interface: usize,
-    reinstall_protocol_interface: usize,
-    uninstall_protocol_interface: usize,
+    install_protocol_interface: unsafe extern "efiapi" fn(
+        handle: NonNull<Option<Handle>>,
```

Comment:
Using `NonNull` here feels a little unusual, although not technically incorrect. Maybe use `&mut Option<Handle>`  instead?

(I realize we don't have very clear standards about how FFI stuff should be declared, I plan to file an issue about that shortly.)